libbpfgo.go

```diff
@@ -1068,6 +1069,31 @@ func (it *BPFMapIterator) Err() error {
 	return it.err
 }
 
+// BPFLinkReader read data from a BPF link
+type BPFLinkReader struct {
+	l  *BPFLink
+	fd int
+}
+
+func (l *BPFLink) Reader() (*BPFLinkReader, error) {
+	fd, errno := C.bpf_iter_create(C.int(l.FileDescriptor()))
+	if fd < 0 {
+		return nil, fmt.Errorf("failed to create reader: %w", errno)
+	}
+	return &BPFLinkReader{
+		l:  l,
+		fd: int(uintptr(fd)),
+	}, nil
+}
+
+func (i *BPFLinkReader) Read(p []byte) (n int, err error) {
+	return syscall.Read(i.fd, p)
+}
+
+func (i *BPFLinkReader) Close() error {
+	return syscall.Close(i.fd)
+}
+
```

Comment:
In the future, if other types require Read and Close syscalls, we must make it them private helpers to be called from the receiver methods. 👍🏼